components/ui/common/form/datePicker/index.tsx

```
@@ -9,32 +9,65 @@ import setMinutes from "date-fns/setMinutes";
-  const startingDate = initialStartDate
-    ? new Date(initialStartDate)
+}: DatePickerComponentProps) {
+  const parsedStartDate = initialStartDate instanceof Date ? initialStartDate : initialStartDate ? new Date(initialStartDate) : null;
+  const parsedEndDate = initialEndDate instanceof Date ? initialEndDate : initialEndDate ? new Date(initialEndDate) : null;
+
+  const startingDate = parsedStartDate && !isNaN(parsedStartDate.getTime())
+    ? parsedStartDate
     : setHours(setMinutes(new Date(), 0), 9);
-  const endingDate = initialEndDate
-    ? new Date(initialEndDate)
+  const endingDate = parsedEndDate && !isNaN(parsedEndDate.getTime())
```

Biome (1.9.4)

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)